lib/index.js

```diff
@@ -27,6 +27,16 @@ var GTM = (module.exports = integration("Google Tag Manager")
     '<script src="//www.googletagmanager.com/gtm.js?id={{ containerId }}&l=dataLayer&gtm_preview={{ environment }}">'
   ));
 
+const buildResetObjects = (props) => {
+  const resetPreviousProps = {};
+  Object.keys(props).forEach((key) => {
+    if (key !== "gtm" && key !== "userConsents") {
+      resetPreviousProps[key] = null;
```

Comment:
How are data appearing in the datalayer then? Still there, or not visible at all?

I remember seeing an advice recommending to use `undefined` instead:
```suggestion
      resetPreviousProps[key] = undefined;
```

If you tried it and it works, don't worry with this!